From 8c5a8add5d3b84bc8b29ec0336ee2f8c0f9ebbff Mon Sep 17 00:00:00 2001
From: "Tan, Raymond" <raymond.tan@intel.com>
Date: Wed, 20 Nov 2019 01:40:05 +0800
Subject: [PATCH 5/7] pwm: pwm-dwc: Fix the parameter type in __dwc_configure

Both duty_cyle and period of the pwm_state struct are of type
unsigned int, and using the wrong parameter type of the function
 __dwc_configure will cause the unwanted type casting.

The type casting can cause issue on DIV_ROUND_CLOSEST calls as well.

Signed-off-by: Tan, Raymond <raymond.tan@intel.com>
---
 drivers/pwm/pwm-dwc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pwm/pwm-dwc.c b/drivers/pwm/pwm-dwc.c
index 027d0f5..30a1789 100644
--- a/drivers/pwm/pwm-dwc.c
+++ b/drivers/pwm/pwm-dwc.c
@@ -70,8 +70,8 @@ static inline void dwc_writel(void __iomem *base, u32 offset, u32 value)
 	writel(value, base + offset);
 }
 
-static void __dwc_configure(struct dwc_pwm *dwc, int pwm, int duty_ns,
-		int period_ns)
+static void __dwc_configure(struct dwc_pwm *dwc, int pwm, unsigned int duty_ns,
+		unsigned int period_ns)
 {
 	u32 ctrl;
 	u32 high;
-- 
2.7.4

